examples/demo/main/main.c

```diff
@@ -49,25 +49,31 @@ static EventGroupHandle_t s_wifi_event_group;
 static const char *TAG = "demo";
 static int s_retry_num = 0;
 static wireguard_config_t wg_config = ESP_WIREGUARD_CONFIG_DEFAULT();
+static wireguard_peer_config_t peer_config = ESP_WIREGUARD_PEER_CONFIG_DEFAULT();
+static uint8_t wireguard_peer_index[WG_MAX_PEERS] = {WIREGUARD_INVALID_PEER};
 
-static esp_err_t wireguard_setup(wireguard_ctx_t* ctx)
+
+static esp_err_t wireguard_setup(wireguard_ctx_t *ctx)
 {
     esp_err_t err = ESP_FAIL;
 
     ESP_LOGI(TAG, "Initializing WireGuard.");
+    // To remove!! These values are pure fictional!
     wg_config.private_key = CONFIG_WG_PRIVATE_KEY;
     wg_config.listen_port = CONFIG_WG_LOCAL_PORT;
-    wg_config.public_key = CONFIG_WG_PEER_PUBLIC_KEY;
-    if (strcmp(CONFIG_WG_PRESHARED_KEY, "") != 0) {
-        wg_config.preshared_key = CONFIG_WG_PRESHARED_KEY;
-    } else {
-        wg_config.preshared_key = NULL;
-    }
-    wg_config.allowed_ip = CONFIG_WG_LOCAL_IP_ADDRESS;
-    wg_config.allowed_ip_mask = CONFIG_WG_LOCAL_IP_NETMASK;
-    wg_config.endpoint = CONFIG_WG_PEER_ADDRESS;
-    wg_config.port = CONFIG_WG_PEER_PORT;
-    wg_config.persistent_keepalive = CONFIG_WG_PERSISTENT_KEEP_ALIVE;
+    wg_config.base_ip = CONFIG_WG_LOCAL_IP_ADDRESS;
+    wg_config.net_mask = CONFIG_WG_LOCAL_IP_NETMASK;
+    peer_config.public_key = CONFIG_WG_PEER_PUBLIC_KEY;
+    // if (strcmp(CONFIG_WG_PRESHARED_KEY, "") != 0) {
+    //     wg_config.preshared_key = CONFIG_WG_PRESHARED_KEY;
+    // } else {
+    peer_config.preshared_key = NULL;
```

Comment:
Maybe it is worth adding second peer and try to ping it as well ..? Just from a smoke testing perspective :smile: